src/mlx_omni_server/rerank/rerank_service.py

```
@@ -0,0 +1,196 @@
+        """Get or load a reranker model based on its ID."""
+        if model_id not in self._models:
+            logger.info(f"Loading reranker model: {model_id}")
+            try:
+                # Resolve model_id to local filesystem path (handles HuggingFace IDs)
+                model_path = resolve_model_path(model_id)
+
+                # Create appropriate reranker based on architecture
+                reranker = create_reranker(model_id, model_path)
+                self._models[model_id] = reranker
+
+            except Exception as e:
```

Ruff (0.14.7)

Do not catch blind exception: `Exception`

(BLE001)